### PR TITLE
Sample material dialog

### LINKED ISSUE #30477: Reinstate Sample Material dialog that was present in MantidPlot
There used to be a right click option on the workspace in MantidPlot called "Sample Material" that displayed this window:

image

Spencer has asked if this can be reinstated in workbench

### PR DESCRIPTION
**Pull request by thomashampson**
**Description of work.**

Reinstated the sample material dialog, which was present in Mantid Plot but not Workbench. It is accessed via an action on the context menu of the workspace widget, but only when a single workspace is selected. It has been rewritten as a python widget.

**To test:**

Load a workspace, right click on it, and click the "Sample material..." action in the context menu. It should display the values stored in the workspace's sample material. Use the "set" button to set the values of the sample material, and the "copy" button to copy the sample material to another workspace.
The context menu action should only appear if a single workspace is selected.
The conetxt menu action should NOT appear if the worksapce doesn't inherit from ExperimentInfo, because then it would not contain a sample object.

Fixes #30477. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by StephenSmith25 — COMMENTED** [review]
Looks great, functions well and adds tests 👍. Just some minor comments and just a small comment that the '...' at the end looks a bit out of place in the menu when compared with other options
image


But I see that quite a few of the plot options have this ellipsis at the end so maybe its just justified. I guess its to signify that more options are available..

File: qt/applications/workbench/workbench/plugins/workspacewidget.py [hunk 1]
```diff
@@ -275,6 +277,28 @@ def _do_show_detectors(self, names):
...
+        # selected. The context menu should only add the sample material action
+        # if there's only one workspace selected.
+        if len(names) == 1:
+            try:
+                workspace = self._ads.retrieve(names[0])
+                presenter = SampleMaterialDialogPresenter(workspace, parent=self)
+                presenter.show_view()
+            except Exception as exception:
+                logger.warning("Could not show sample material for workspace "
+                               "'{}':\n{}\n".format(workspace.name(), exception))
```
[the remark this answers is not in the document]
Comment (thomashampson) [reply qt/applications/workbench/workbench/plugins/workspacewidget.py]: Good spot!

File: qt/python/mantidqt/widgets/samplematerialdialog/test/test_samplematerial_presenter.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def tearDown(self):
+        AnalysisDataService.clear()
+
+    def test_sample_material_update(self):
+        """
+        Check that the sample material values in the view are up to date.
+        The view should have been updated when the presenter was created.
+        """
+        assert(self.presenter.view.formula_item.text(1) == self.formula)
```
[the remark this answers is not in the document]
Comment (thomashampson) [reply qt/python/mantidqt/widgets/samplematerialdialog/test/test_samplematerial_presenter.py]: Done

**Comment by thomashampson** [comment]
Changed the name of the context menu action to "Show Sample Material" so that it's consistent with the rest of them.

**Review by StephenSmith25 — APPROVED** [review]
Great, good work 👍!

**Pull request merged**